client/src/app/site/agenda/services/agenda-repository.service.ts

```diff
@@ -49,7 +48,7 @@ export class AgendaRepositoryService extends BaseRepository<ViewItem, Item> {
      *
      * TODO: used over not-yet-existing detail view
      */
-    public update(item: Partial<Item>, viewUser: ViewItem): Observable<Item> {
```

Comment:
There shouldn't be any Promises returned in the Repositories. Observables are needed by the Front-End Functions and in the Front-End shouldn't be any promises. Also: Auto-Updates are likely to not work on a Promise-Based systems, because you don't receive automatic updates (like in observables) on changes of the data-set.